app/Controllers/AddDonationController.php

```diff
@@ -242,4 +238,23 @@ private function resetSessionState(): void {
 			false
 		);
 	}
+
+	/**
+	 * @param ConstraintViolation[] $constraintViolations
+	 */
+	private function logValidationErrors( $constraintViolations ): void {
+		$formattedConstraintViolations = [];
+		foreach ( $constraintViolations as $constraintViolation ) {
```

Comment:
What, no `array_map`? 😛 😸

Reply:
I was also thinking that. This way you avoid the $formattedConstraintViolations variable that gets mutated and have the string formatting not be in the same function as the call to the logger (different levels of abstraction). Decided to not comment on it since the current code is hardly bad.